org.eclipse.xtext/src/org/eclipse/xtext/build/ClusteringStorageAwareResourceLoader.java

```diff
@@ -54,33 +47,14 @@ public <T> Iterable<T> executeClustered(Iterable<URI> uris, Function1<? super Re
 				loadedURIsCount = 0;
 			}
 			loadedURIsCount++;
-			if (isSource(uri)) {
-				sourceLevelURIs.add(uri);
-				Resource existingResource = resourceSet.getResource(uri, false);
-				if (existingResource instanceof StorageAwareResource) {
-					if (((StorageAwareResource) existingResource).isLoadedFromStorage()) {
-						existingResource.unload();
-					}
-				}
-				SourceLevelURIsAdapter.setSourceLevelUrisWithoutCopy(resourceSet, sourceLevelURIs);
-			}
 			resources.add(resourceSet.getResource(uri, true));
 		}
 		FluentIterable.from(resources).transform(operation::apply).copyInto(result);
 		return result;
 	}
 
 	/**
-	 * Return true if the given uri must be loaded from source.
-	 */
-	protected boolean isSource(URI uri) {
```

Comment:
I wonder if we should keep this method here for backwards compatiblity reasons and delegate to it from the newly introduced isSource on the IncrementalBuilder

Reply:
I was also wondering as well if we should keep it. I have left the method as it was, just a @Deprecated tag. 
It would be possible to depend on the IncrementalBuilder, but one would need to create a new instance of it, call setContext on the new instance before one could call the method. It looks a bit too much for my taste, but I could do that if you prefer.

Reply:
I'm mainly concerned about code that *might* have overwritten this method assuming that it would be called. @cdietrich what's your take? Remove it completely to be sure that we *break* the client rather than silently offering an internal API that's no longer in use?

Reply:
we can remove it with the next release